### ISSUE TITLE
Atlas incorrectly switched styling for light & dark stickers.

### ISSUE THREAD

**Issue by ethib137** [liferay/clay#4310]
In atlas, stickers change the color of the text/icon when the type changes. Atlas incorrectly set the text/icon color for sticker-light to $dark and the text/icon color for sticker-dark to $light.

https://github.com/liferay/clay/blob/aa4653f901d22386711b2807c501e9e94a2a1dd5/packages/clay-css/src/scss/atlas/variables/_stickers.scss#L64
```
$sticker-success-color: $success !default; // reference
$sticker-light-color: $dark !default; // should be $white
$sticker-dark-color: $white !default; // should be $dark
```

In the clay base css the stickers change the background-color when the type changes. In base light and dark are handled correctly.

https://github.com/liferay/clay/blob/aa4653f901d22386711b2807c501e9e94a2a1dd5/packages/clay-css/src/scss/variables/_stickers.scss#L88
```
$sticker-success-bg: $success !default;
$sticker-light-bg: $light !default;
$sticker-dark-bg: $dark !default;
```

This may be a difficult change to make since there are a number of places in Liferay that depend on the incorrect styles. It would be necessary to simultaneous update those places in Liferay when the fixed version of clay was added. 

This is not a high priority for me, but I figured it should still be reported.

**Comment by pat270** [comment]
@ethib137 Thanks for reporting this, we could do `$sticker-light-color: color-yiq($light);` and `$sticker-dark-color: color-yiq($dark);`. It shouldn't cause any issues.

**Comment by ethib137** [comment]
@pat270 that would still result in the same issue.. color-yiq($light) will still render a dark color and color-yiq($dark) will still render a light color.

The pattern that is being set for all other stickers is this:
sticker-primary will render a primary colored icon
sticker-success will render a success colored icon

But then with light and dark the pattern is incorrectly applied:
sticker-light will render a dark colored icon
sticker dark will render a light colored icon

Do you see what I mean?

**Comment by pat270** [comment]
@ethib137 putting it that way makes it more clear. It makes sense since the sticker's main color is the foreground color. There was an update in v2.9 where Lexicon swapped the foreground and background colors for stickers. I never updated light and dark.

This would be a breaking change. From my quick search of portal there are only 5 files that use `sticker-light` and 0 files that use `sticker-dark`. I'm not sure how we would inform the community of this breaking change if we implement it. A blog post, breaking changes section, or all of the above?

**Comment by matuzalemsteles** [comment]
@pat270 yeah, I think the best way would be to create a gradual deprecation of this and remove it completely during a major version and we'll be accumulating this information to put in a future v3 to v4 migration session, a blog post explaining this plan would be nice.

**Comment by pat270** [comment]
@matuzalemsteles So we are only deprecating and not changing the values?

**Comment by matuzalemsteles** [comment]
Yeah, I what we can do here is maybe added a flag that changes this behavior now and in a v4 version, we can move that by default. We still have to check if that would be possible too, I imagine that's works.

**Comment by pat270** [comment]
@matuzalemsteles I don't think that is worth it. It's easy to swap the colors via variables for light and dark stickers. Our options should be break it or leave it. Breaking it doesn't seem that bad to me (it is a bug) and we actually "broke" it in v2.9 and I haven't heard anything negative about it 😅 .

**Comment by matuzalemsteles** [comment]
@pat270 Well, I understand. The downside is that changing now again is that maybe people think the bug is expected behavior and trust it now. But if this isn't a radical change, we can think about changing it otherwise I'm not sure it would be safe to do this now.

**Comment by pat270** [comment]
Either way is good with me, I think we can leave it and close this as won't fix. If we get more requests we can reopen and fix it?

**Comment by nhpatt** [comment]
I would break it being a bug that we broke in a previous version.

**Comment by ethib137** [comment]
Just to provide an example... One place this is used is in segmentation.

Here is how the stickers are currently rendered:
![CurrentBehavior](https://user-images.githubusercontent.com/1609196/135963831-a6909870-d94c-4e45-8d74-3df0438ba155.png)

And here is how they would be rendered after fixing clay:
![ProposedFix](https://user-images.githubusercontent.com/1609196/135963844-b4dfde99-4aa8-4995-ac7f-c2285d2d4e9f.png)

This doesn't look horrible, but it definitely looks unintended.

Wouldn't it be very doable to simply find and swap all instances of sticker-light and sticker-dark when releasing the fixed version of clay?

**Comment by nhpatt** [comment]
I think it's doable and we could even leave it that way after talking with the product teams. I would fix it, as it's a bug :D

**Comment by matuzalemsteles** [comment]
Well, we can just check all the places that are used inside the DXP and analyze, if we've broken it in the past and people haven't identified it maybe fix now they won't notice it was broken 🤷‍♂️

**Comment by pat270** [comment]
My [research](https://github.com/liferay/clay/issues/4310#issuecomment-930376900) says there are 5 files that use `sticker-dark`. We just need to change them to `sticker-light`.

**Comment by julien** [comment]
@pat270 yes, that's why I'm seeing too. Do you want me to send a pull request for that?

**Comment by pat270** [comment]
@julien sure it sounds like we're all on board.

**Comment by julien** [comment]
Great

**Comment by ethib137** [comment]
> My [research](https://github.com/liferay/clay/issues/4310#issuecomment-930376900) says there are 5 files that use `sticker-dark`. We just need to change them to `sticker-light`.

Don't forget to also change the `sticker-light` places to `sticker-dark`. ;)

**Comment by pat270** [comment]
I got it backwards, edited my comment.

**Comment by julien** [comment]
@ethib137 yes of course. I've just had a call with @pat270 via slack, once the change is done here, I'll send a pull request for both.
